src/azul/plugins/repository/tdr_hca/__init__.py

```diff
@@ -290,6 +293,13 @@ def _parse_drs_uri(self,
         else:
             return None
 
+    def _validate_drs_uri(self, drs_uri: Optional[str]) -> None:
+        # These requirements prevent mismatches in the DRS domain, and ensure
+        # that changes to the column syntax don't go undetected.
+        if drs_uri is not None:
+            drs_uri = RegularDRSURI.parse(drs_uri)
+            require(drs_uri.netloc == config.tdr_service_url.netloc)
```

Comment:
It appears that this method was pushed down from TDRBundle but no abstract method declaration was added. Also, can the conditional be moved to the call site? The possibility of the argument being None complicates the signature and requires every implementation to check for it.

Reply:
> It appears that this method was pushed down from TDRBundle but no abstract method declaration was added. 

These methods are never called via the superclass interface. They're independent implementation details of the two subclasses. We could choose to inline one of them, or change its name or signature, and it wouldn't affect anything outside of the class in which it's defined. I think adding a superclass stub falsely implies otherwise.

> Also, can the conditional be moved to the call site? The possibility of the argument being None complicates the signature and requires every implementation to check for it.

It can be done, but the only reason these methods exist at all is to avoid having that sort of clutter at the call sites. Currently, each method has exactly one call site, so literally the only benefit would be removing `Optional` from the signature.

Now that I think of it, I'm convinced the AnVIL implementation needs this check at all. In HCA, there are phantom files without DRS URI's, but this is allowed for AnVIL?

Reply:
Decided at PL to just inline the methods.